packages/@n8n/nodes-langchain/nodes/vector_store/VectorStoreQdrant/VectorStoreQdrant.node.ts

```diff
@@ -0,0 +1,138 @@
+		default: {},
+		options: [
+			embeddingDimensions,
+			{
+				displayName: 'Is Auto Embedded',
+				name: 'isAutoEmbedded',
+				type: 'boolean',
+				default: true,
+				description: 'Whether to automatically embed documents when they are added',
+			},
+		],
+	},
+];
+
+const retrieveFields: INodeProperties[] = [
+	{
+		displayName: 'Options',
+		name: 'options',
+		type: 'collection',
+		placeholder: 'Add Option',
+		default: {},
+		options: [embeddingDimensions, metadataFilterField],
+	},
+];
+
+export const VectorStoreQdrant = createVectorStoreNode({
+	meta: {
+		displayName: 'Qdrant Vector Store',
+		name: 'vectorStoreQdrant',
+		description: 'Work with your data in Qdrant Vector Store',
+		credentials: [
+			{
+				name: 'qdrantApi',
+				required: true,
+			},
+		],
+		icon: 'file:qdrant.png',
+		docsUrl:
+			'https://docs.n8n.io/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.vectorstoreqdrant/',
+	},
+	sharedFields: [
+		{
+			displayName: 'Collection Name',
+			name: 'collectionName',
+			type: 'string',
+			default: '',
+			required: true,
+		},
+	],
+	insertFields,
+	loadFields: retrieveFields,
+	retrieveFields,
+	async getVectorStoreClient(context, filter, embeddings, itemIndex) {
+		const collectionName = context.getNodeParameter('collectionName', itemIndex) as string;
+
+		const options =
+			(context.getNodeParameter('options', itemIndex) as {
+				embeddingDimensions?: number;
+			}) || {};
+
+		const credentials = (await context.getCredentials('qdrantApi')) as {
+			apiKey?: string;
+			apiUrl: string;
+		};
+		const parts = url.parse(credentials.apiUrl);
+
+		const qdrantConfig: QdrantLibArgs = {
+			collectionName,
+			collectionConfig: {
+				vectors: {
+					size: options.embeddingDimensions ?? 1536,
+					distance: 'Cosine',
+				},
+			},
+			client: new QdrantClient({
+				apiKey: credentials.apiKey,
+				host: parts.hostname ?? undefined,
+				port: Number.parseInt(parts.port ?? ''),
+			}),
+		};
+
+		return new QdrantVectorStore(embeddings, qdrantConfig);
+	},
+	async populateVectorStore(context, embeddings, documents, itemIndex) {
+		const collectionName = context.getNodeParameter('collectionName', itemIndex) as string;
+		const options =
+			(context.getNodeParameter('options', itemIndex) as {
+				isAutoEmbedded?: boolean;
+				embeddingDimensions?: number;
+			}) || {};
+
+		const credentials = (await context.getCredentials('qdrantApi')) as {
+			apiKey?: string;
+			apiUrl: string;
+		};
+
+		const parts = url.parse(credentials.apiUrl);
+		const qdrantConfig: QdrantLibArgs = {
+			collectionName,
+			collectionConfig: {
+				vectors: {
+					size: options.embeddingDimensions ?? 1536,
+					distance: 'Cosine',
+				},
+			},
+			client: new QdrantClient({
+				apiKey: credentials.apiKey,
+				host: parts.hostname ?? undefined,
+				port: Number.parseInt(parts.port ?? ''),
+			}),
+		};
+
+		QdrantVectorStore.fromDocuments(documents, embeddings, qdrantConfig) as Promise<VectorStore>;
```

Comment:
We've recently did some changes to improve handling of these vector store operations. Could you please merge in latest `ai-beta` and implement similar error handling as we have for `VectorStoreSupabase` or `VectorStorePinecone`? I'm also wondering if we should add an option to create a collection if it's missing. WDYT?